src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +37,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+
+    this->dyna.actor.shape.rot.y += this->unk_210;
+
+    Math_ApproxF(&this->unk_208, 50.0f, 1.5f);
+    Math_ApproxS(&this->unk_20C, 400, 15);
+
+    this->unk_20E += this->unk_20C;
+
+    this->dyna.actor.posRot.pos.x = (Math_Sins(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.x;
+    this->dyna.actor.posRot.pos.z = (Math_Coss(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.z;
+
+    if (this->colliderJntSph.base.acFlags & 2) {
+        colliderBaseAc = this->colliderJntSph.base.ac;
+
+        if (colliderBaseAc != NULL) {
+            positionDiff = colliderBaseAc->posRot.pos.y - this->dyna.actor.posRot.pos.y;
+
+            if (positionDiff > 120.0f && positionDiff < 200.0f) {
+                if (Math3D_Dist2DSq(colliderBaseAc->posRot.pos.z, this->colliderJntSph.base.ac->posRot.pos.x,
+                                    this->dyna.actor.posRot.pos.z, this->dyna.actor.posRot.pos.x) < SQ(32.0f)) {
+                    func_800800F8(globalCtx, 4210, 240, this, 0);
+                    func_808B7D38(this);
+                    func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                }
+            }
+        }
+    }
+    func_8002F974(this, NA_SE_EV_ELEVATOR_MOVE - SFX_FLAG);
+}
+
+void func_808B7D38(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7D50;
+    this->unk_216 = 0;
+    this->unk_214 = 0;
+}
+
+void func_808B7D50(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    f32 tempValue2;
+    f32 tempValue;
+
+    if (this->unk_216 > 120) {
+        Math_ApproxS(&this->unk_210, 0x3E8, 0x32);
+    } else {
+        Math_ApproxS(&this->unk_210, 0xBB8, 0x64);
+    }
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.shape.rot.y + this->unk_210;
+
+    if (this->unk_216 < 70) {
+        Math_ApproxF(&this->unk_208, 100.0f, 2.0f);
+    } else {
+        Math_ApproxF(&this->unk_208, 0.0f, 2.0f);
+    }
+
+    Math_ApproxS(&this->unk_20C, 1000, 20);
+
+    this->unk_20E += this->unk_20C;
+
+    this->dyna.actor.posRot.pos.x = (Math_Sins(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.x;
+    this->dyna.actor.posRot.pos.z = (Math_Coss(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.z;
+
+    this->unk_212 += 0xBB8;
+
+    Math_ApproxS(&this->unk_214, 0x5DC, 0x1E);
+
+    this->dyna.actor.shape.rot.x = Math_Coss(this->unk_212) * this->unk_214;
+    this->dyna.actor.shape.rot.z = -Math_Sins(this->unk_212) * this->unk_214;
+
+    if (this->unk_216 > 140) {
+        func_808B7F74(this);
+    }
+
+    if (this->unk_216 < 80) {
+        func_808B7770(this, globalCtx, 1.0f);
+    } else {
+        func_808B7770(this, globalCtx, 0.8f);
+    }
+
+    tempValue2 = (this->unk_210 - 500) * 0.0006f;
+
+    tempValue = CLAMP(tempValue2, 0.0f, 1.5f);
+
+    func_800F436C(&this->dyna.actor.projectedPos, NA_SE_EV_WALL_MOVE_SP - SFX_FLAG, tempValue);
+}
+
+void func_808B7F74(BgSpot18Basket* this) {
```

Comment:
Rotations are normally in hex.

Reply:
I wish we wouldn't enforce hex vs dec for these, since both styles can make perfect sense depending on the values used (and it's probably not worth looking for and bringing up in reviews, if we want to be slightly more efficient).

Also we technically never agreed on hex in the first place, to be clear.

Reply:
for now I think hex is better personally, but I agree that its something we need to decide and enforce later. it shouldnt slow us down